lineman/app/components/thread_page/comment_form/comment_form.coffee

```diff
@@ -1,35 +1,40 @@
 angular.module('loomioApp').directive 'commentForm', ->
-  scope: {comment: '='}
+  scope: {discussion: '='}
   restrict: 'E'
   templateUrl: 'generated/components/thread_page/comment_form/comment_form.html'
   replace: true
-  controller: ($scope, $rootScope, FlashService, Records, CurrentUser, KeyEventService, AbilityService, ScrollService) ->
-    discussion = $scope.comment.discussion()
-    group = $scope.comment.group()
+  controller: ($scope, $rootScope, FormService, Records, CurrentUser, KeyEventService, AbilityService, ScrollService) ->
 
     $scope.formInFocus = ->   $rootScope.$broadcast 'hideFeedbackForm'
     $scope.formLostFocus = -> $rootScope.$broadcast 'showFeedbackForm'
 
     $scope.showCommentForm = ->
-      AbilityService.canAddComment(discussion)
+      AbilityService.canAddComment($scope.discussion)
 
     $scope.threadIsPublic = ->
-      discussion.private == false
+      $scope.discussion.private == false
 
     $scope.threadIsPrivate = ->
-      discussion.private == true
-
-    $scope.submit = ->
-      $scope.isDisabled = true
-      $scope.comment.save().then ->
-        $scope.isDisabled = false
-        if $scope.comment.isReply()
-          FlashService.success('comment_form.messages.replied', name: $scope.comment.parent().authorName())
-        else
-          FlashService.success('comment_form.messages.created')
-        $scope.comment = Records.comments.build(discussionId: discussion.id)
-      , ->
-        $scope.isDisabled = false
+      $scope.discussion.private == true
+
+    successMessage = ->
+      if $scope.comment.isReply()
+        'comment_form.messages.replied'
+      else
+        'comment_form.messages.created'
+    successMessageName = ->
+      if $scope.comment.isReply()
+        $scope.comment.parent().authorName()
+
+    $scope.init = ->
+      $scope.comment = Records.comments.build(discussionId: $scope.discussion.id)
```

Comment:
This pattern solves the issue of FormService not updating its model dynamically.

personally I don't think the model of the form should be able to change while the user is completing it, so I'm happy with this as a way to reuse the same form again.